Gemfile

```diff
@@ -48,6 +45,15 @@ unless Gem::Version.new(Bundler::VERSION) >= Gem::Version.new('1.5.0')
   exit 1
 end
 
+def optional
+  group :optional, &proc
+end
+
+# Let optional gems optional
```

Comment:
Optional gems should be:

1. easily loadable as a group from config/application.rb without having to list every group name
2. easily excludable individually (Note: a gem is excluded only if all groups it belongs to are specified in `--without`)

To satisfy 1 and 2, I put them all in a common group named `:optional` and then set it to the "without" list by default.